src/time-range-picker/time-range-picker.tsx

```diff
@@ -0,0 +1,220 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { TimeRangePickerHelper } from "./helper";
+import { TimeRangePickerDropdown } from "./timerangepicker-dropdown";
+import {
+    ArrowRangeIcon,
+    ArrowRight,
+    BottomHighlightEndTime,
+    BottomHighlightStartTime,
+    InputSelectorEndTimeElement,
+    InputSelectorStartTimeElement,
+    TimeContainer,
+    Wrapper,
+} from "./timerangepicker.styles";
+import { TimeRangePickerProps } from "./types";
+
+export const TimeRangePicker = ({
+    id,
+    disabled = false,
+    error,
+    value,
+    defaultValue,
+    placeholder,
+    format = "24hr",
+    readOnly,
+    onChange,
+    onBlur,
+    onSelectionCancel,
+    ...otherProps
+}: TimeRangePickerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showStartTimeSelector, setShowStartTimeSelector] =
+        useState<boolean>(false);
+    const [showEndTimeSelector, setShowEndTimeSelector] =
+        useState<boolean>(false);
+    const [startTimeVal, setStartTimeVal] = useState("");
+    const [endTimeVal, setEndTimeVal] = useState("");
+
+    const nodeRef = useRef<HTMLDivElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        if (value) {
+            setStartTimeVal(value.startTime);
+            setEndTimeVal(value.endTime);
+        }
+    }, []);
+    useEffect(() => {
+        document.addEventListener("mousedown", handleMouseDownEvent);
+        document.addEventListener("keyup", handleKeyUpEvent);
```

Comment:
noticed some inconsistent behaviour with Shift+Tab

https://github.com/LifeSG/react-design-system/assets/16853062/3daca06c-ba88-4a5c-9a9c-7e7a31a72614

it might be easier to use `onBlur` to manage focus events